src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java

```diff
@@ -0,0 +1,54 @@
+package com.spotify.dns;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.xbill.DNS.*;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NoCachingLookupFactoryTest {
```

Comment:
Im trying to find ways to automate this. A quick way i found to test this is basically asking for resolution of a name twice, and before the second run, disabling network in the workstation.

Using `SimpleLookupFactory` it does not fail, given that uses the Lookup cache.
Using `NoCachingLookupFactory` fails, given that is trying to resolve again.